src/ui/public/saved_object_table/item_table_state.js

```diff
@@ -0,0 +1,38 @@
+import { SortOrder } from 'ui_framework/components/table/sort_order';
+
+export class ItemTableState {
```

Comment:
@weltenwort After more thought, I think where we both end up in terms of concrete suggestions is very similar (as you point out in comparing `calculateItemsOnPage` with your suggestion). I'm just a bit leery about proscribing a series of concrete steps like you did... it's easy to end up following a process like that blindly, but it's important that at each step, we make sure that it is actually getting us closer to making the code simpler.

This is why I would suggest first finding useful abstractions, etc. and then applying your recipe to it. If the code is already in a simple state when we apply your recipe, then it's easier to gauge whether we are introducing unnecessary complexity.